src/core/lib/security/transport/client_auth_filter.cc

```diff
@@ -165,9 +165,17 @@ static void on_credentials_metadata(void* arg, grpc_error* input_error) {
     grpc_metadata_batch* mdb =
         batch->payload->send_initial_metadata.send_initial_metadata;
     for (size_t i = 0; i < calld->md_array.size; ++i) {
-      add_error(&error, grpc_metadata_batch_add_tail(
-                            mdb, &calld->md_links[i],
-                            GRPC_MDELEM_REF(calld->md_array.md[i])));
+      // Only add x-goog-user-project header if not present.
```

Comment:
This does not belong here.  The client_auth_filter code should be agnostic of any individual credential type, so it should have no knowledge of specific metadata keys like the x-goog-user-project header.  All knowledge of that key should be contained in the google_default_creds implementation.

Reply:
See discussion below.